src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUploadServiceApplicationDocuments.java

```diff
@@ -94,16 +94,34 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
         var caseData = details.getData();
         AlternativeService alternativeService = caseData.getAlternativeService();
 
-        State endState = AwaitingServiceConsideration;
-
-        if (StringUtils.isNotEmpty(alternativeService.getServicePaymentFee().getHelpWithFeesReferenceNumber())) {
-            endState = AwaitingServicePayment;
-        }
+        State endState = determineEndState(alternativeService, details.getState());
 
         return AboutToStartOrSubmitResponse.<CaseData, State>builder()
             .data(caseData)
             .state(endState)
             .build();
     }
+
+    private State determineEndState(final AlternativeService alternativeService, State currentState) {
+        State endState = currentState;
+
+        if (!AwaitingDocuments.equals(endState)) {
+            return endState;
+        }
+
+        endState = State.AwaitingServiceConsideration;
+        if (alternativeService.getServiceApplicationSubmittedOnline() != null
+            && alternativeService.getServiceApplicationSubmittedOnline().toBoolean()) {
+            if (StringUtils.isNotEmpty(alternativeService.getServicePaymentFee().getHelpWithFeesReferenceNumber())) {
+                endState = AwaitingServicePayment;
+            }
+        } else {
+            if (alternativeService.getAlternativeServiceFeeRequired().toBoolean()) {
+                endState = AwaitingServicePayment;
+            }
+        }
+
+        return endState;
+    }
```

Comment:
We could refactor this method to this providing it doesn't break the original functionality? 
```
private State determineEndState(final AlternativeService alternativeService, State currentState) {
    if (!AwaitingDocuments.equals(currentState)) {
        return currentState;
    } else if ((alternativeService.getServiceApplicationSubmittedOnline() != null
                && alternativeService.getServiceApplicationSubmittedOnline().toBoolean() 
                && StringUtils.isNotEmpty(alternativeService.getServicePaymentFee().getHelpWithFeesReferenceNumber()))
               || alternativeService.getAlternativeServiceFeeRequired().toBoolean()) {
        return AwaitingServicePayment;
    } else {
        return State.AwaitingServiceConsideration;
    }
}
```